src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceDeserializer.cs

```diff
@@ -1061,136 +1062,2239 @@ internal ODataJsonLightReaderNestedInfo ReadPropertyWithoutValue(IODataJsonLight
+                case ODataAnnotationNames.ODataType:
+
+                    // TODO: skip the odata.type;
+                    await this.ReadAndValidateAnnotationStringValueAsync(ODataAnnotationNames.ODataType)
+                        .ConfigureAwait(false);
+                    break;
+                default:
+                    ODataAnnotationNames.ValidateIsCustomAnnotationName(annotationName);
+                    Debug.Assert(
+                        !this.MessageReaderSettings.ShouldSkipAnnotation(annotationName),
+                        "!this.MessageReaderSettings.ShouldReadAndValidateAnnotation(annotationName) -- otherwise we should have already skipped the custom annotation and won't see it here.");
+                    object instanceAnnotationValue = await this.ReadCustomInstanceAnnotationValueAsync(propertyAndAnnotationCollector, annotationName)
+                        .ConfigureAwait(false);
+                    resourceSet.InstanceAnnotations.Add(new ODataInstanceAnnotation(annotationName, instanceAnnotationValue.ToODataValue()));
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously reads resource property which doesn't have value, just annotations.
+        /// </summary>
+        /// <param name="resourceState">The state of the reader for resource to read.</param>
+        /// <param name="propertyName">The name of the property read.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains a reader nested resource info representing 
+        /// the nested resource info detected while reading the resource contents,
+        /// or null if no nested resource info was detected.
+        /// </returns>
+        /// <remarks>
+        /// Pre-Condition:  JsonNodeType.EndObject              The end of the resource object.
+        ///                 JsonNodeType.Property               The property after the one we're to read.
+        /// Post-Condition: JsonNodeType.EndObject              This method doesn't move the reader.
+        ///                 JsonNodeType.Property
+        /// </remarks>
+        internal async Task<ODataJsonLightReaderNestedInfo> ReadPropertyWithoutValueAsync(IODataJsonLightReaderResourceState resourceState, string propertyName)
+        {
+            Debug.Assert(resourceState != null, "resourceState != null");
+            Debug.Assert(!string.IsNullOrEmpty(propertyName), "!string.IsNullOrEmpty(propertyName)");
+            this.AssertJsonCondition(JsonNodeType.Property, JsonNodeType.EndObject);
+
+            ODataJsonLightReaderNestedInfo readerNestedInfo = null;
+            IEdmStructuredType resourceType = resourceState.ResourceType;
+            IEdmProperty edmProperty = this.ReaderValidator.ValidatePropertyDefined(propertyName, resourceType);
+            if (edmProperty != null && !edmProperty.Type.IsUntyped())

```

Comment:
nit: If we inverted the conditions and returned from each block we could reduce the indentation :). not a blocker.

Reply:
Fixed